brion/synapse.cpp

```diff
@@ -357,20 +395,17 @@ class Synapse : public boost::noncopyable
         lunchbox::ScopedWrite mutex( detail::_hdf5Lock );
         SilenceHDF5 silence;
 
-        BOOST_FOREACH( UnmappedFiles::value_type& entry, _unmappedFiles )
+        // this trial-and-error is the 'fastest' path found
+        BOOST_FOREACH( const std::string& candidate, _fileNames )
         {
-            const std::string& candidate = entry.first;
-            H5::H5File& file = entry.second;
-            // keeping the files open 'saves' some time
```

Comment:
Yes, indeed it seemed like it did ;-)
It might be too much work to do this now, but I think this modification would have deserved to be in a separate commit with this explanation in the commit message. It helps for future reference and for the code review to know why things have been modified.